drivers/sensor/max31865/max31865.c

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright (c) 2022 HAW Hamburg FTZ-DIWIP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "max31865.h"
+
+static int max31865_spi_write(const struct device *dev, uint8_t reg, uint8_t *data, size_t len)
+{
+	const struct max31865_config *cfg = dev->config;
+
+	const struct spi_buf bufs[] = {{
+					       .buf = &reg,
+					       .len = 1,
+				       },
+				       {.buf = data, .len = len}};
+
+	const struct spi_buf_set tx = {.buffers = bufs, .count = 2};
+
+	return spi_write_dt(&cfg->spi, &tx);
+}
+
+static int max31865_spi_read(const struct device *dev, uint8_t reg, uint8_t *data, size_t len)
+{
+	const struct max31865_config *cfg = dev->config;
+
+	reg &= 0x7F;
+	const struct spi_buf tx_buf = {.buf = &reg, .len = 1};
+	const struct spi_buf_set tx = {.buffers = &tx_buf, .count = 1};
+	struct spi_buf rx_buf[] = {{
+					   .buf = &reg,
+					   .len = 1,
+				   },
+				   {.buf = data, .len = len}};
+	const struct spi_buf_set rx = {.buffers = rx_buf, .count = 2};
+
+	return spi_transceive_dt(&cfg->spi, &tx, &rx);
+}
+
+/**
+ * @brief Set device configuration register
+ *
+ * @param device device instance
+ * @return 0 if successful, or negative error code from SPI API
+ */
+static int configure_device(const struct device *dev)
+{
+	struct max31865_data *data = dev->data;
+	uint8_t cmd[] = {data->config_control_bits};
+	int err = max31865_spi_write(dev, WR(REG_CONFIG), cmd, 1);
+
+	if (err < 0) {
+		LOG_ERR("Error write SPI%d\n", err);
+	}
+	return err;
+}
+
+/**
+ * @brief Set device fail threshold registers
+ *
+ * @param device device instance
+ * @return 0 if successful, or negative error code from SPI API
+ */
+static int set_threshold_values(const struct device *dev)
+{
+	const struct max31865_config *config = dev->config;
+	uint8_t cmd[] = {
+		(config->high_threshold >> 7) & 0x00ff, (config->high_threshold << 1) & 0x00ff,
+		(config->low_threshold >> 7) & 0x00ff, (config->low_threshold << 1) & 0x00ff};
+	int err = max31865_spi_write(dev, WR(REG_HIGH_FAULT_THR_MSB), cmd, 4);
+
+	if (err < 0) {
+		LOG_ERR("Error write SPI%d\n", err);
+	}
+	return err;
+}
+
+#ifdef CONFIG_NEWLIB_LIBC
```

Comment:
With the introduction of picolibc, I think this can be `CONFIG_FULL_LIBC`

Reply:
I can't find that Kconfig symbol anywhere, only `CONFIG_REQUIRES_FULL_LIBC`. Is that what you meant?